draft-ietf-privacypass-architecture.md

```diff
@@ -107,24 +82,72 @@ of all participating entities.
+previously authenticated server action, without having to reauthenticate
+manually. The tokens retain anonymity in the sense that the act of
+revealing them cannot be linked back to the session where they were
+initially issued.
+
+At a high level, Privacy Pass is composed of two protocols: issuance
+and redemption. Issuance is a protocol between three logical entities:
+Client, Mediator, and Issuer. The Issuer is responsible for issuing
+tokens in response to requests from Clients. The Mediator is responsible
+for attesting properties about the Client for which tokens are issued.
+For example, in the original Privacy Pass protocol {{PPSRV}}, tokens were
+only issued to Clients that solved CAPTCHAs. In this context, the Mediator
+attested that some client solved a CAPTCHA and the resulting token
+produced by the Issuer was proof of this fact. Depending on the information
+being attested, Mediators may also store state about individual Clients,
+such as the number of overall tokens issued thus far.
+
+The redemption protocol runs between Client and Origin (server). It allows
+Origins to challenge Clients to present one or more tokens for authorization.
+Depending on the type of token, e.g., whether or not it is cross-origin
+or per-origin, and whether or not it can be cached, the Client either presents
+a previously obtained token or invokes the issuance protocol to acquire one
+for authorization.
+
+The issuance and redemption protocols operate in concert as shown in
+the figure below.
+
+~~~
+ Client        Mediator          Issuer          Origin
+
+    <---------------------------------------- Challenge \
+                                                        |
++--------------------------------------------\          |
+| TokenRequest --->                          |          |
+|                    TokenRequest --->       |          |
+|                                 (validate) | Issuance | Redemption
+|                                 (evaluate) |   Flow   |   Flow
+|                    <--- TokenResponse      |          |
+|   <--- TokenResponse                       |          |
+---------------------------------------------/          |
+                                                        |
+     Response -------------------------------------- >  /
+~~~
+{: #fig-overview title=" Privacy Pass Architectural Components"}
+
+This document describes requirements for both issuance and redemption
+protocols. This document also describes ecosystem considerations that
+impact the stated privacy and security guarantees of the protocol.
+For instance, the number of servers issuing Privacy Pass tokens, along
+with the number of registered clients, determines the anonymity set of
+each individual client. Moreover, this can be influenced by other factors,
+such as the key rotation policy used by each server. There are also client
+behavior patterns that can reduce the effective security of the server.
+Additional considerations include:
```

Comment:
This and the following points all seem unnecessary in the Intro, given the preceding sentence that says that the document will describe ecosystem considerations.